### PR TITLE
#1329 Refactor TextSourceConverter to use supplier to get InputStreamReader

### PR DESCRIPTION
**Pull request by eechih**
### Checklist
- [x] [Review Contribution Guidelines](https://github.com/oharastream/ohara/blob/master/CONTRIBUTING.md)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#authors))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #1329)
- [x] Add unit tests (or please explain why this PR is not worth having new tests)
- [x] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-kafka/src/main/java/com/island/ohara/kafka/connector/text/TextFileSystem.java [hunk 1]
```diff
@@ -33,13 +34,12 @@
...
-   * create a inputStreamReader for input file. NOTE: this resource will be automatically closed by
-   * the {@link TextSourceTask#_poll()}
+   * create a supplier of InputStreamReader for the input file
    *
    * @param path a full path form input file
-   * @return a inputStreamReader from input file
+   * @return a supplier of InputStreamReader
    */
-  InputStreamReader createReader(String path);
+  Supplier<InputStreamReader> createReader(String path);
```
Comment (chia7712) [inline ohara-kafka/src/main/java/com/island/ohara/kafka/connector/text/TextFileSystem.java]: Lazy creation, normally, is suitable to be a input type than returned type. The **time** to create object is up to caller.

Comment (eechih) [reply ohara-kafka/src/main/java/com/island/ohara/kafka/connector/text/TextFileSystem.java]: Understand, thanks for reminder. ^_^

Comment (eechih) [reply ohara-kafka/src/main/java/com/island/ohara/kafka/connector/text/TextFileSystem.java]: @chia7712 I fixed this issue. Could you review again? Thanks :)

File: ohara-kafka/src/main/java/com/island/ohara/kafka/connector/text/csv/CsvSourceConverter.java [hunk 2]
```diff
@@ -44,13 +47,17 @@
...
-    List<RowSourceRecord> records = toRecords(rowsAndIndex);
-    // ok. all data are prepared. let update the cache
-    rowsAndIndex.keySet().forEach(index -> cache.update(path, index));
-    return records;
+  public List<RowSourceRecord> convert(Supplier<InputStreamReader> supplier) {
+    try (InputStreamReader reader = supplier.get()) {
+      Map<Integer, List<Cell<String>>> cellsAndIndex = toCells(reader);
+      Map<Integer, Row> rowsAndIndex = transform(cellsAndIndex);
+      List<RowSourceRecord> records = toRecords(rowsAndIndex);
+      // ok. all data are prepared. let update the cache
```
Comment (konekoya) [inline ohara-kafka/src/main/java/com/island/ohara/kafka/connector/text/csv/CsvSourceConverter.java]: _Let's_ :)

Comment (eechih) [reply ohara-kafka/src/main/java/com/island/ohara/kafka/connector/text/csv/CsvSourceConverter.java]: Thanks :)

**Pull request merged**